src/VecSim/algorithms/svs/svs_tiered.h

```diff
@@ -700,18 +716,53 @@ class TieredSVSIndex : public VecSimTieredIndex<DataType, float> {
         }();
 
         if (label_exists) {
-            std::scoped_lock lock(this->flatIndexGuard, this->journal_mutex);
-            if (this->frontendIndex->isLabelExists(label)) {
-                ret = this->frontendIndex->deleteVector(label);
-                assert(ret == 1 && "unexpected deleteVector result");
-                journal.emplace_back(label, false);
+            std::lock_guard lock(this->flatIndexGuard);
+
+            auto deleting_ids = this->frontendIndex->getElementIds(label);
+            // assert if all elements of deleting_ids are unique
+            assert(std::set<idType>(deleting_ids.begin(), deleting_ids.end()).size() ==
+                       deleting_ids.size() &&
+                   "deleting_ids should contain unique ids");
+
+            if (!deleting_ids.empty()) {
+                // Sort deleting_ids by id descending order
+                std::sort(deleting_ids.begin(), deleting_ids.end(),
+                          [](const auto &a, const auto &b) { return a > b; });
+
+                // Delete vector from the frontend index.
+                auto updated_ids = this->frontendIndex->deleteVectorAndGetUpdatedIds(label);
+                // Record swaps in the journal.
+                for (auto id : deleting_ids) {
+                    auto it = updated_ids.find(id);
+                    if (it != updated_ids.end()) {
+                        assert(id == it->first &&
+                               "id in updated_ids should match the id in deleting_ids");
+                        auto newId = id;
+                        auto oldId = it->second.first;
+                        auto oldLabel = it->second.second;
+                        this->swaps_journal.emplace_back(oldLabel, oldId, newId);
```

Comment:
Let's try to cover this edge case with a unit test, I believe we have a similar test in HNSW

Reply:
Added tests in test_svs_tiered.cpp called `testSwapJournalSingle` and `testSwapJournalMulti`.
Tried to emulate as worst cases as possible.